File: antson.java/src/test/java/io/odysz/anson/jprotocol/AnSessionResp.java
package io.odysz.anson.jprotocol;

import io.odysz.anson.jprotocol.AnsonMsg;
import io.odysz.anson.jprotocol.AnsonResp;

public class AnSessionResp extends AnsonResp {

	SessionInf ssInf;

	public AnSessionResp(AnsonMsg<AnsonResp> parent, String ssid, String uid, String ... roleId) {
		super(parent);
		ssInf = new SessionInf(ssid, uid, roleId == null || roleId.length == 0 ? null : roleId[0]);
		ssInf.ssid = ssid;
		ssInf.uid = uid;
//		if (roleId != null && roleId.length > 0)
//			ssInf.roleId = roleId[0];
	}

	public AnSessionResp(AnsonMsg<? extends AnsonResp> parent, SessionInf ssInf) {
		super(parent);
		this.ssInf = ssInf;
	}

	public AnSessionResp() {
		super("");
	}

	public SessionInf ssInf() {
		return ssInf;
	}
}

File: antson.csharp/antson.csharp/semantics/io.odysz/jsession/AnSessionResp.cs
using io.odysz.semantic.jprotocol;

namespace io.odysz.semantic.jsession
{
	public class AnSessionResp : AnsonResp
	{
		public SessionInf ssInf { get; protected set; }

		public AnSessionResp(AnsonMsg parent, string ssid, string uid, params string[] roleId)
			: base(parent)
		{
			ssInf = new SessionInf(ssid, uid, roleId == null || roleId
				.Length == 0 ? null : roleId[0]);
			//ssInf.ssid = ssid;
			//ssInf.uid = uid;
		}

		public AnSessionResp(AnsonMsg parent, SessionInf ssInf)
			: base(parent)
		{
			// TODO built-in role?
			// if (roleId != null && roleId.length > 0)
			// 	ssInf.roleId = roleId[0];
			this.ssInf = ssInf;
		}

		public AnSessionResp() : base(string.Empty)
		{
		}
	}
}
